code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -9,13 +9,21 @@
 	amount_per_transfer_from_this = 5
 	possible_transfer_amounts = list(1,2,3,4,5)
 	volume = 5
-	var/filled = 0
+
+/obj/item/weapon/reagent_containers/dropper/on_reagent_change()

```

Comment:
I'll re-iterate what I said elsewhere.

Is there any reason that what you have under `update_icon` isn't just here, instead? With `on_reagent_change` just calling another proc, you're just adding additional proc call overhead.